.evergreen/config_generator/components/funcs/backtrace.py

```diff
@@ -0,0 +1,34 @@
+from shrub.v3.evg_command import FunctionCall
+
+from config_generator.etc.utils import bash_exec
+
+
+class Backtrace:
+    @classmethod
+    def name(cls):
+        return 'backtrace'
+
+    @classmethod
+    def defn(cls):
+        commands = []
+
+        commands.append(
+            bash_exec(
+                working_dir='mongoc',
+                script='.evergreen/scripts/debug-core-evergreen.sh',
+            )
+        )
+
+        return {cls.name(): commands}
+
+    @classmethod
+    def call(cls, **kwargs):
+        return FunctionCall(func=cls.name(), **kwargs)
```

Comment:
These classes all have a common shape. I recommend a common base class:

```py
class FunctionDef:
    name: ClassVar[str]
    commands: Sequence[EvgCommand]

    @classmethod 
    def call(cls, **kwargs):
        return FunctionCall(cls.name, **kwargs)
    
    @classmethod 
    def defn(cls) -> Mapping[str, Sequence[EvgCommand]]:
        return {cls.name: list(cls.commands)}
        
...

class Backtrace(FunctionDef):
    name = 'backtrace'
    commands = [
        bash_exec(working_dir='mongoc', script='.evergreen/scripts/debug-core-evergreen.sh')
    ]
```

Reply:
`classmethod` will respect inheritance overriding, so you can leave the `call` on the base class, and just redefine `call` on derived classes, if appropriate (the `cls` parameter refers to the actual derived class given as the left-hand of the `.` operator).

Reply:
> while avoiding warnings due to differing parameters in overridden class functions

PyLint will generate the following warnings:

> Number of parameters was 2 in 'Function.call' and is now 3 in overridden 'FetchBuild.call' methodpylint(arguments-differ)

> Variadics removed in overridden 'FetchBuild.call' methodpylint(arguments-differ)

Reply:
There are ways to get those warnings quiet, but probably more work than it's worth here.